python/cudf/cudf/core/_internals/where.py

```diff
@@ -156,14 +156,14 @@ def _normalize_columns_and_scalars_type(
 
             source_df = frame.copy(deep=False)
             others = []
-            for col_name, other_sclr in zip(frame._column_names, other):
+            for i, col_name in enumerate(frame._column_names):
 
                 (
                     source_col,
                     other_scalar,
                 ) = _check_and_cast_columns_with_other(
                     source_col=source_df._data[col_name],
-                    other=other_sclr,
+                    other=other[i],
```

Comment:
Now that we are eliminating the requirement for `other` to be iterable, would it make sense to maybe use a flag to track whether or not we want to use a scalar here? Something like:

```python
other=(other if cudf.utils.dtypes.is_scalar(other) else other[i])
```

Asking because the list comp on line 155 seems a little redundant now.

Reply:
Yup, made this optimization 👍